source/core/CEntryPoints.cpp

```diff
@@ -118,67 +118,92 @@ VIREO_EXPORT Int32 EggShell_PokeMemory(TypeManagerRef tm,
+        return kEggShellResult_ObjectNotFoundAtPath;
 
-    WriteDoubleToMemory(actualType, pData, d);
+    return kEggShellResult_Success;
 }
 //------------------------------------------------------------
-//! Read a numeric value from a symbol. Value will be coerced as needed.
-VIREO_EXPORT Double EggShell_ReadDouble(TypeManagerRef tm, const char* viName, const char* eltName)
+//! Get a reference to the type pointer and data for a sub element
+VIREO_EXPORT EggShellResult EggShell_FindSubValue(TypeManagerRef tm,
+        const TypeRef typeRef, void * pData, const char* eltName, TypeRef* typeRefLocation, void** dataRefLocation)
 {
-    void *pData = nullptr;
-    SubString objectName(viName);
+    if (typeRef == nullptr || !typeRef->IsValid())
+        return kEggShellResult_InvalidTypeRef;
+
+    TypeManagerScope scope(tm);
     SubString path(eltName);
-    TypeRef actualType = tm->GetObjectElementAddressFromPath(&objectName, &path, &pData, true);
-    if (actualType == nullptr)
-        return -1;
+    *typeRefLocation = typeRef->GetSubElementAddressFromPath(&path, pData, dataRefLocation, true);
+    if (*typeRefLocation == nullptr)
+        return kEggShellResult_ObjectNotFoundAtPath;
 
-    return ReadDoubleFromMemory(actualType, pData);
+    return kEggShellResult_Success;
 }
 //------------------------------------------------------------
-// Write a string value to a symbol. Value will be parsed according to format designated.
-VIREO_EXPORT void EggShell_WriteValueString(TypeManagerRef tm,
-        const char* viName, const char* eltName, const char* format, const char* value)
+//! Write a numeric value to a symbol. Value will be coerced as needed.
+VIREO_EXPORT EggShellResult EggShell_WriteDouble(TypeManagerRef tm, const TypeRef typeRef, void* pData, Double value)
 {
     TypeManagerScope scope(tm);
+    if (typeRef == nullptr || !typeRef->IsValid())
+        return kEggShellResult_InvalidTypeRef;
 
-    void *pData = nullptr;
+    NIError error = WriteDoubleToMemory(typeRef, pData, value);
+    if (error)
+        return kEggShellResult_UnexpectedObjectType;
+    return kEggShellResult_Success;
+}
+//------------------------------------------------------------
+//! Read a numeric value from a symbol. Value will be coerced as needed.
+VIREO_EXPORT EggShellResult EggShell_ReadDouble(TypeManagerRef tm, const TypeRef typeRef, const void* pData, Double* result)
+{
+    TypeManagerScope scope(tm);
+    if (typeRef == nullptr || !typeRef->IsValid())
+        return kEggShellResult_InvalidTypeRef;
+
+    if (result == nullptr)
+        return kEggShellResult_InvalidResultPointer;
+
+    NIError error = kNIError_Success;
+    *result = ReadDoubleFromMemory(typeRef, pData, &error);
+    if (error)
+        return kEggShellResult_UnexpectedObjectType;
+    return kEggShellResult_Success;
+}
+//------------------------------------------------------------
+// Write a string value to a symbol. Value will be parsed according to format designated.
```

Comment:
"to designated format" instead of "to format designated"?

Reply:
I'll change it